app/controllers/events_controller.rb

```diff
@@ -1,7 +1,7 @@
 class EventsController < ApplicationController
   def index
     @keyword = params["search"]
-    @events = @keyword.present? ? Event.search(@keyword) : Event.upcoming
+    @events = @keyword.present? ? Event.search(@keyword) : Event.upcoming.includes(:venue, :category)
```

Comment:
It's not a good idea to inject the `includes` in `Event.seach`. What if you want to use the search somewhere else and that place doesn't need to load `venue` and `category` along? This practice could lead to wasting preloading when project gets more complex. So I'm thinking of
```ruby
 @events = (@keyword.present? ? Event.search(@keyword) : Event.upcoming).includes(:venue, :category)
```